### PR TITLE
Restructure Günstigerprüfung

### PR DESCRIPTION
**Pull request by ChristianZimpelmann**
### What problem do you want to solve?

Restructure Günstigerprüfung between Kindergeld and Kinderfreibetrag as follows:

- Make sure that for single parents income tax gain by Kinderfreibetrag is compared to only half of Kindergeld (although they receive the full Kindergeld)
- Do not set Kindergeld to 0 if Kinderfreibetrag is beneficial. Instead, Kindergeld is added to the income tax. [Source](https://www.gesetze-im-internet.de/estg/__2.html#:~:text=(4)%20Der%20Gesamtbetrag%20der%20Eink%C3%BCnfte,au%C3%9Fergew%C3%B6hnlichen%20Belastungen%2C%20ist%20das%20Einkommen.): "Wird das Einkommen in den Fällen des § 31 um die Freibeträge nach § 32 Absatz 6 gemindert, ist der Anspruch auf Kindergeld nach Abschnitt X der tariflichen Einkommensteuer hinzuzurechnen" 

### Todo

- [x] Implement changes as described above
- [x] Make sure that those are correct
- [x] Adjust test cases
- [x] Check whether any tutorials need to be adjusted?
- [x] If it is a relevant contribution, put it in `CHANGES.rst`.

### CODE REVIEW

File: gettsim/transfers/kinderbonus.py [hunk 1]
```diff
@@ -8,17 +7,14 @@ def kinderbonus_m(
...
-    kinderfreib_günstiger_tu
-        See :func:`kinderfreib_günstiger_tu`.
     kinderbonus_basis_m
         See :func:`kinderbonus_basis_m`.
 
     Returns
     -------
 
     """
-    out = 0.0 if kinderfreib_günstiger_tu else kinderbonus_basis_m
```
Comment (Eric-Sommer) [inline gettsim/transfers/kinderbonus.py:20]: It seems that `kinderbonus_m` can be removed and `kinderbonus_basis_m` can be renamed to `kinderbonus_m`. Same for kindergeld

File: gettsim/taxes/eink_st.py [hunk 2]
```diff
@@ -135,29 +138,57 @@ def eink_st_tu_ab_1997(
 
 def kinderfreib_günstiger_tu(
```
Comment (Eric-Sommer) [inline gettsim/taxes/eink_st.py:139]: Is this function still used?

**Review by Eric-Sommer — COMMENTED** [review]
Thanks a lot! Adding kindergeld to the income tax instead of reducing the benefit should simplify lots of things.

File: gettsim/taxes/eink_st.py (hunk 2 above)
Comment (ChristianZimpelmann) [reply gettsim/taxes/eink_st.py:139]: Yes, in `eink_st_tu_ab_1997`.

File: gettsim/transfers/kinderbonus.py (hunk 1 above)
Comment (ChristianZimpelmann) [reply gettsim/transfers/kinderbonus.py:20]: Yes, absolutely!

**Review by ChristianZimpelmann — COMMENTED** [review]
Thanks! Just one more comment.

Two comments by @mjbloemer (@LauraGergeleit, would be great if you can add them):
- ich würde in den code documentation von `eink_st_ohne_kinderfreib_tu` und `eink_st_mit_kinderfreib_tu` noch reinschreiben, was die "tarifliche ESt I" und was die "tarifliche ESt II" ist. Das sind scheinbar festehende Begriffe und die Info ist hilfreich. (ich würde die Variablen aber so lassen)
- wird abhängig von `if kinderfreib_günstiger_tu` nochmal das zu versteuernde Einkommen irgendwo festgelegt? (-> please add a variable `zu_verst_eink_tu`)

I also added the contribution to `CHANGES.md` and adjusted the tutorials slightly to not contradict the new implementation of Günstigerprüfung. For more substantial changes, see #474.

**Review by ChristianZimpelmann — COMMENTED** [review]
Thank you, @LauraGergeleit for the last additions!

I would approve the PR, but cannot do so as I started it originally. Could you do it, @LauraGergeleit?

**Review by LauraGergeleit — APPROVED** [review]

**Pull request merged**